storage.go

```diff
@@ -25,6 +25,7 @@ type StorageEngine interface {
 	DeleteById(collectionName string, id string) error
 	IterateDocs(q *Query, consumer docConsumer) error
 	Insert(collection string, docs ...*Document) error
+	Save(collection string, doc *Document) error
```

Comment:
The StorageEngine interface should contain only basic primitives. In this case, the **Save** method can be implemented using Insert() and Update(), so no need to add a new method on the StorageEngine interface.